tools/integration_tests/operations/parallel_dirops_test.go

```diff
@@ -0,0 +1,447 @@
+		defer wg.Done()
+		fileInfo, err := os.Stat(filePath)
+		return fileInfo, err
+	}
+	var stat1, stat2 os.FileInfo
+	var err1, err2 error
+
+	// Parallel lookups of file just under mount.
+	filePath := path.Join(testDir, "file1.txt")
+	wg := sync.WaitGroup{}
+	wg.Add(2)
+	go func() {
+		stat1, err1 = lookUpFunc(&wg, filePath)
+	}()
+	go func() {
+		stat2, err2 = lookUpFunc(&wg, filePath)
+	}()
+	wg.Wait()
+
+	// Assert both stats passed and give correct information
+	assert.NoError(t, err1)
+	assert.NoError(t, err2)
+	assert.Equal(t, int64(5), stat1.Size())
+	assert.Equal(t, int64(5), stat2.Size())
+	assert.Contains(t, filePath, stat1.Name())
+	assert.Contains(t, filePath, stat2.Name())
+
+	// Parallel lookups of file under a directory in mount.
+	filePath = path.Join(testDir, "explicitDir1/file2.txt")
+	wg.Add(2)
+	go func() {
+		stat1, err1 = lookUpFunc(&wg, filePath)
+	}()
+	go func() {
+		stat2, err2 = lookUpFunc(&wg, filePath)
+	}()
+	wg.Wait()
+
+	// Assert both stats passed and give correct information
+	assert.NoError(t, err1)
+	assert.NoError(t, err2)
+	assert.Equal(t, int64(10), stat1.Size())
+	assert.Equal(t, int64(10), stat2.Size())
+	assert.Contains(t, filePath, stat1.Name())
+	assert.Contains(t, filePath, stat2.Name())
+}
+
+func TestParallelReadDirs(t *testing.T) {
+	// Create directory structure for testing.
+	testDir := createDirectoryStructureForParallelDiropsTest(t)
+	readDirFunc := func(wg *sync.WaitGroup, dirPath string) ([]os.DirEntry, error) {
+		defer wg.Done()
+		dirEntries, err := os.ReadDir(dirPath)
+		return dirEntries, err
+	}
+	var dirEntries1, dirEntries2 []os.DirEntry
+	var err1, err2 error
+
+	// Parallel readDirs of explicit dir under mount.
+	dirPath := path.Join(testDir, "explicitDir1")
+	wg := sync.WaitGroup{}
+	wg.Add(2)
+	go func() {
+		dirEntries1, err1 = readDirFunc(&wg, dirPath)
+	}()
+	go func() {
+		dirEntries2, err2 = readDirFunc(&wg, dirPath)
+	}()
+	wg.Wait()
+
+	// Assert both readDirs passed and give correct information
+	assert.NoError(t, err1)
+	assert.NoError(t, err2)
+	assert.Equal(t, 2, len(dirEntries1))
+	assert.Equal(t, 2, len(dirEntries2))
+	assert.Contains(t, "file1.txt", dirEntries1[0].Name())
+	assert.Contains(t, "file2.txt", dirEntries1[1].Name())
+	assert.Contains(t, "file1.txt", dirEntries2[0].Name())
+	assert.Contains(t, "file2.txt", dirEntries2[1].Name())
+
+	// Parallel readDirs of a directory and its parent directory.
+	dirPath = path.Join(testDir, "explicitDir1")
+	parentDirPath := testDir
+	wg = sync.WaitGroup{}
+	wg.Add(2)
+	go func() {
+		dirEntries1, err1 = readDirFunc(&wg, dirPath)
+	}()
+	go func() {
+		dirEntries2, err2 = readDirFunc(&wg, parentDirPath)
+	}()
+	wg.Wait()
+
+	// Assert both readDirs passed and give correct information
+	assert.NoError(t, err1)
+	assert.NoError(t, err2)
+	assert.Equal(t, 2, len(dirEntries1))
+	assert.Equal(t, 4, len(dirEntries2))
+	assert.Contains(t, "file1.txt", dirEntries1[0].Name())
```

Comment:
Just curious, why are we not able to use Equal?

Reply:
just didn't want to know the whole path and then compare.

Reply:
Didn't get. Anyway not a blocker to merge the PR.